src/MusicPlaylistKernel.java

```diff
@@ -0,0 +1,60 @@
+import components.map.Map;
+import components.standard.Standard;
+
+/**
+ * Music playlist kernel component with primary methods.
+ *
+ * @initially <pre>
+ * ():
+ *  ensures
+ *   this = {}
+ * </pre>
+ */
+public interface MusicPlaylistKernel extends Standard<MusicPlaylist> {
+
+        /**
+         * Adds a singular song to the playlist, given as a Map.Pair.
+         *
+         * @param song
+         *                Map.Pair<String, String> that will be added to
+         *                playlist
+         */
+        void add(Map.Pair<String, String> song);
```

Comment:
Make sure you include design by contract elements like `@requires` and `@ensures`